dart/lib/src/utils/tracing_utils.dart

```diff
@@ -0,0 +1,36 @@
+import '../../sentry.dart';
+
+void addSentryTraceHeader(ISentrySpan span, Map<String, dynamic> headers) {
+  final traceHeader = span.toSentryTrace();
+  headers[traceHeader.name] = traceHeader.value;
+}
+
+void addBaggageHeader(ISentrySpan span, Map<String, dynamic> headers) {
+  final baggage = span.toBaggageHeader();
+  if (baggage != null) {
+    // TODO: overwrite if the sentry key already exists
+    // https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/#baggage
+    var currentValue = headers[baggage.name] as String? ?? '';
+
+    if (currentValue.isNotEmpty == true) {
+      currentValue = '$currentValue,';
+    }
+    currentValue = '$currentValue${baggage.value}';
+
+    headers[baggage.name] = currentValue;
+  }
+}
+
+bool containsTracePropagationTarget(
+    List<String> tracePropagationTargets, String url) {
+  if (tracePropagationTargets.isEmpty) {
+    return true;
+  }
+  for (final target in tracePropagationTargets) {
+    final regExp = RegExp(target, caseSensitive: false);
+    if (url.contains(target) || regExp.hasMatch(url)) {
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
`tracePropagationTargets` impl. reference.

Reply:
In the Java SDK the regex is not case insensitive. Is this something we should change in the Java SDK?

Reply:
In the Java SDK the regex is not case insensitive. Should we change it on the Java SDK?

Reply:
@adinauer I would say we should.